runtime/development/src/lib.rs

```diff
@@ -978,28 +978,54 @@ impl pallet_migration_manager::Config for Runtime {
 	type WeightInfo = weights::pallet_migration_manager::SubstrateWeight<Self>;
 }
 
-// our base filter
-// allow base system calls needed for block production and runtime upgrade
-// other calls will be disallowed
-pub struct BaseFilter;
-
-impl Contains<Call> for BaseFilter {
+/// Base Call Filter
+/// We block any call that could lead for tranche tokens to be transferred through XCM.
+pub struct BaseCallFilter;
+impl Contains<Call> for BaseCallFilter {
 	fn contains(c: &Call) -> bool {
-		matches!(
-			c,
-			// Calls from Sudo
-			Call::Sudo(..)
-			// Calls for runtime upgrade
-			| Call::System(frame_system::Call::set_code{..})
-			| Call::System(frame_system::Call::set_code_without_checks{..})
-			// Calls that are present in each block
-			| Call::ParachainSystem(
-				cumulus_pallet_parachain_system::Call::set_validation_data{..}
-			)
-			| Call::Timestamp(pallet_timestamp::Call::set{..})
-			// Claiming logic is also enabled
-			| Call::CrowdloanClaim(pallet_crowdloan_claim::Call::claim_reward{..})
-		)
+		match c {
+			Call::PolkadotXcm(method) => match method {
+				// We disable all PolkadotXcm extrinsics that allow users to build XCM messages
+				// from scratch, which could have them transferring Tranche tokens.
+				// To transfer tokens, use XTokens, for which we have specific filters
+				// blocking tranche transfers.
+				// To send a raw XCM message, use orml_xcm, which ensures the origin of
+				// such call to be root or majority of the collective.
+				pallet_xcm::Call::send { .. }
+				| pallet_xcm::Call::execute { .. }
+				| pallet_xcm::Call::teleport_assets { .. }
+				| pallet_xcm::Call::reserve_transfer_assets { .. }
+				| pallet_xcm::Call::limited_reserve_transfer_assets { .. }
+				| pallet_xcm::Call::limited_teleport_assets { .. } => {
+					return false;
+				}
+				pallet_xcm::Call::force_xcm_version { .. }
+				| pallet_xcm::Call::force_default_xcm_version { .. }
+				| pallet_xcm::Call::force_subscribe_version_notify { .. }
+				| pallet_xcm::Call::force_unsubscribe_version_notify { .. } => {
+					return true;
+				}
+				pallet_xcm::Call::__Ignore { .. } => {
+					unimplemented!()
+				}
+			},
+			Call::XTokens(method) => match method {
+				orml_xtokens::Call::transfer {
+					currency_id: CurrencyId::Tranche(_, _),
+					..
+				}
+				| orml_xtokens::Call::transfer_with_fee {
+					currency_id: CurrencyId::Tranche(_, _),
+					..
+				}
+				// We preemptively disable this as we haven't encountered an use case for it.
+				// Shall some user or use case require it, we will make it more fine-grained.
+				| orml_xtokens::Call::transfer_multicurrencies { .. } => false,
+				// Any other XTokens call is good to go
```

Comment:
What about `transfer_multiasset`, this can't be used to send tranche tokens then?

Reply:
Users could use that call to transfer tranche tokens, that's why we block it. As I left commented, we have never come across the need to use this call neither do I see the need for it in the foreseeable future so we block it.

Reply:
The blocked call is `transfer_multicurrencies` though, while my question is about `transfer_multiasset` :)

Reply:
Thank you, good catch! I missed a few more actually. Fixed 👍